classes/stock/StockAvailable.php

```diff
@@ -353,6 +357,44 @@ public static function setProductOutOfStock($id_product, $out_of_stock = false,
         }
     }
 
+    /**
+     * @param int $id_product
+     * @param string $location
+     * @param int $id_shop Optional
+     * @param int $id_product_attribute Optional
+     *
+     * @return bool
+     *
+     * @throws PrestaShopDatabaseException
+     */
+    public static function setLocation($id_product, $location, $id_shop = null, $id_product_attribute = 0)
+    {
+        if (false === Validate::isUnsignedId($id_product)) {
```

Comment:
We can do better here:

```php
if (
    false === Validate::isUnsignedId($id_product)
    || false === Validate::isUnsignedId($id_shop)
   || false === Validate::isUnsignedId($id_product_attribute)
   || false === Validate::isString($location)
) {
    throw new SomeSpecializedException('Some meaningful message with serialized data passed');
}
```

And you can remove almost all cast operations you're doing, also this will remove unexpected weird behaviors that can come from wrong data passed to this function.